The Todo list app

adamclasic opened the pull request:
The Todo list app creates todos in separate lists. The app loads with a default project to which all of the todos are initially put. The user is able to create new projects and choose which project their todos go into.

This project should prove the understanding of the student on the following JavaScript concepts:

Object Oriented Programming (OOP)
Factory Functions
The Module Pattern
Single Responsibility and Tightly Coupled Objects

adamclasic committed:
add feature branch

adamclasic committed:
add package.json

adamclasic committed:
add webpack config

adamclasic committed:
add styles files

adamclasic committed:
add aside bar

adamclasic committed:
add style to side

adamclasic committed:
add background color

adamclasic committed:
add style to content section

adamclasic committed:
add font awsome

adamclasic committed:
add font awsome

adamclasic committed:
add task factory function

adamclasic committed:
add completeTask

adamclasic committed:
add event listner to tasks

adamclasic committed:
add trash icon

adamclasic committed:
fix removeTask

adamclasic committed:
add validations to form

adamclasic committed:
add projects render

adamclasic committed:
add create project window

adamclasic committed:
add add project

adamclasic committed:
add remove project

adamclasic committed:
add addProject to dom

adamclasic committed:
add removeProject to dom

adamclasic committed:
add trash on on hover

adamclasic committed:
add render tasts on project click

adamclasic committed:
add projects/task relations

adamclasic committed:
add capitlization to project name

adamclasic committed:
add import/export

adamclasic committed:
fix deleting last project error

adamclasic committed:
add Local storage

adamclasic committed:
fix console log errors

adamclasic committed:
add animation to task

adamclasic committed:
fix LoacalStorage not updating upon task edit

adamclasic committed:
fix animation

adamclasic committed:
add animation on removeTask

adamclasic committed:
update readme

adamclasic committed:
fix some eslint errors

adamclasic committed:
fix render project list

adamclasic committed:
fix some eslint errors

adamclasic committed:
fix some eslint errors

adamclasic committed:
fix unnecissary import

adamclasic committed:
fix stlye lint

adamclasic committed:
update readme

Oghenebrume50 commented on src/scripts/tasks.js:
```diff
@@ -0,0 +1,31 @@
+
+  const addTask = (name) => {
+    if (name.length < 1) { return; }
+    const newTask = {
```
Task should have more properties as shown in the screenshot below 

![see this](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/todo_requirement.png)

Oghenebrume50 commented on src/scripts/project.js:
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable import/no-cycle, import/no-named-default */
+import { renderProjects, renderTask, projectId } from './render';
```
avoid circular dependencies this is usually a pointer to bad software design, see link https://stackoverflow.com/questions/51094117/dependency-cycle-detected-import-no-cycle

Oghenebrume50 reviewed (changes requested):
### CHANGES REQUIRED

Hi @adamclasic you did a very nice work :+1: the UI is great and very straight forward.  :+1: 
I have outlined some changes need below please work on these 

- Todos/tasks should be editable 
![see this](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/ui_requirement.png)

Please **request for another review after changes are made**

**RAPHAEL NORIODE** :mask: STAY_SAFE

adamclasic committed:
fix btn

adamclasic committed:
add localstorage to render.js

adamclasic committed:
update render with mainProjects.render

adamclasic committed:
fix renderTask argument typo

adamclasic committed:
update before merge

adamclasic committed:
fix import cycle

adamclasic committed:
add expand btn style

adamclasic committed:
add edit task modal style

adamclasic committed:
update task.js with editTask

adamclasic committed:
add fill/scrap editTask form

adamclasic committed:
fix import no name issue

adamclasic committed:
add logic to save changes

adamclasic committed:
fix date not showing

adamclasic committed:
update style to edit task modal

adamclasic committed:
add show on hover to edit btn

adamclasic committed:
update edit buton postion to absolut

adamclasic committed:
fix linter errors

adamclasic committed:
update readme

adamclasic committed:
update readme

adamclasic committed:
update readme

adamclasic committed:
update readme

adamclasic commented:
In this Pull Request Review I have:

- fixed the import cycle issue.

- fixed the import No Default Name issue.

- add an expanded task window to edit and view.

Torres-ssf commented on package.json:
```diff
@@ -2,9 +2,10 @@
+  "private": true,
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack"
```
- [x] Nice that you added Webpack to your scripts at the package.json file :+1: , but we need to go further and create at least 2 scripts for your application, one for development and another one for production. These scripts are supposed to inform in which [mode](https://webpack.js.org/configuration/mode/) Webpack will run. We will have something like this:

```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
  },
```

- [x] We also need to inform users about these script in the README file. This information can be placed in the Usage section. This way users will know how they can use your application.

- [x] The test script can be safely deleted since we have no tests implemented.

**OPTIONAL**

- [ ]  You can create an even better script for development stages, using the `
webpack-dev-server` and create a script like this:
```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```
Running the `start` script will automatically open the index file in the browser, without the need to open it manually

Torres-ssf commented on src/index.js:
```diff
@@ -0,0 +1,253 @@
+  dateInput.value = date;
+}
+
+// function scrapEditTaskForm(index) {
+//   mainProjects.projectsArr[projectId].editTask(
+//   index,
+//   titleInput.value,
+//   descriptionInput.value,
+//   dateInput.value,
+//   )
+// }
```
- [x] Since this is supposed to be production code, it's bad practice to leave commented code. We need to delete all commented code.

Torres-ssf commented on src/scripts/render.js:
```diff
@@ -0,0 +1,10 @@
+// export {
+//   renderProjects, renderTask, addEventListenerToTasksForm,
+//   addEventListenerToProjectsForm, projectId,
+// };
```
- [x] Since this is supposed to be a production code, we need to delete all commented code. If the `render.js` has no more purpose in the project, we can safely delete it.

Torres-ssf commented on src/index.js:
```diff
@@ -0,0 +1,253 @@
+//   )
+// }
+
+// testing 123
```
- [x] This is not a commented code but it's not an informative comment. We should also delete this.

Torres-ssf commented on src/index.js:
```diff
@@ -0,0 +1,253 @@
+  });
+}
+
+// testing 123
```
- [x] This is not a commented code but it's not an informative comment. We should also delete this.

Torres-ssf commented on src/index.js:
```diff
@@ -0,0 +1,253 @@
+  return liElemt;
+}
+
+// testing 321
```
- [x] This is not a commented code but it's not an informative comment. We should also delete this.

Torres-ssf reviewed (changes requested):
### Changes required :muscle: 
Hi @adamclasic :hand: 
Great job on this project so far :clap: 
I really liked your UI. It's looking really nice.
But before moving on we still need to make some changes.

- [x] We need to delete `app.js` file since it empty and therefore, it has no purpose in the project.

- [x] There is a bug in your application right now, after editing the last task's name, it also changed the name of almost all other tasks in the process
![Screenshot from 2020-09-02 10-38-01](https://user-images.githubusercontent.com/30977845/91990849-9eebdb80-ed08-11ea-8e99-cd1fb5bef2e8.png)
![Screenshot from 2020-09-02 10-38-23](https://user-images.githubusercontent.com/30977845/91990862-a14e3580-ed08-11ea-8bfe-515991d992fd.png)

Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**
[Twitter](https://twitter.com/torres_ssf) [LinkedIn](https://www.linkedin.com/in/torres-ssf/) [Portifolio](https://www.torres-ssf.com/)

adamclasic committed:
fix edit task issue

adamclasic committed:
update readme

adamclasic committed:
fix eslint errors

bellom reviewed (approved):
## Hi @adamclasic  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**
